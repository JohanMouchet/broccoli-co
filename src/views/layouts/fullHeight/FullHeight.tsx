import React from "react";
import cx from "classnames";
import { Footer, Header, Main } from "views/partials";
import styles from "./FullHeight.module.scss";

type Props = {
  children: React.ReactNode;
};

const FullHeight: React.FC<Props> = ({ children }) => (
  <div className={styles["layout"]}>
    <div className={cx(styles["container"], "container container--xl")}>
      <Header />
      <Main>{children}</Main>
      <Footer />
    </div>
  </div>
);

export default FullHeight;
