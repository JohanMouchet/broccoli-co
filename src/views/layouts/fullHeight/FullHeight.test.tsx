import React from "react";
import { render } from "@testing-library/react";
import FullHeight from "./FullHeight";

it("Renders <FullHeight /> without crashing", () => {
  expect(
    render(<FullHeight>children</FullHeight>).container.firstChild
  ).toMatchSnapshot();
});
