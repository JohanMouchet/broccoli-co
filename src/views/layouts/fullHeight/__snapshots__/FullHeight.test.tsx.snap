// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <FullHeight /> without crashing 1`] = `
<div
  class="layout"
>
  <div
    class="container container container--xl"
  >
    <header
      class="header"
    >
      <span
        class="logo"
      >
        <span
          class="logo-icon"
        >
          <svg>
            logo.svg
          </svg>
        </span>
        <h2
          class="logo-text"
        >
          Broccoli & Co.
        </h2>
      </span>
    </header>
    <main
      class="main"
    >
      children
    </main>
    <footer
      class="footer"
    >
      <span
        class="copy"
      >
        Made with ♡ in 
        Melbourne
        . Developed by
         
        <a
          href="https://www.johan-mouchet.com/"
        >
          Johan Mouchet
        </a>
        .
      </span>
      <i
        class="copy"
      >
        © 
        2020
         
        Broccoli & Co.
         All rights reserved.
      </i>
    </footer>
  </div>
</div>
`;
